### ISSUE TITLE
MSO ASE API support

### ISSUE THREAD

**Issue by rsmeyers** [CiscoDevNet/ansible-mso#113]
We need to add support for MSO ASE platform in the MSO Ansible modules.

Pref. this would be implemented as following:
a) provide a config option to support non-ASE and ASE deployments
b) default to non-ASE deployments
c) if undefined, display a warning saying we are defaulting to non-ASE
d) if we have more ASE then non-ASE in the field, start announcing we will shift the default in x-months trough warnings when being undefined, and then switch

**Comment by manofcolombia** [comment]
Does this collection currently not function with MSO running on CASE? 
We are getting ready to move MSO over to CASE after the next CASE release. That will likely delay our move if so.

**Comment by lhercot** [comment]
Even with older releases MSO on SE was supported and working.

This is fixed in v1.2.0 as part of the MSO HTTPAPI plugin and in the ND HTTPAPI plugin (in cisco.nd collection). 

To use it add the following variables for the MSO HTTPAPI plugin (MSO < v3.2):
```yaml
ansible_connection=ansible.netcommon.httpapi
ansible_network_os=cisco.mso.mso
```
To use it add the following variables for the ND HTTPAPI plugin (MSO >= v3.2):
```yaml
ansible_connection=ansible.netcommon.httpapi
ansible_network_os=cisco.nd.nd
```

See more details in the README.
